src/components/app/details/triggerView/CIMaterialModal.tsx

```diff
@@ -30,6 +42,24 @@ export const CIMaterialModal = ({
     ...props
 }: CIMaterialModalProps) => {
     const { ciNodeId } = useParams<Pick<CIMaterialRouterProps, 'ciNodeId'>>()
+    const context = React.useContext(TriggerViewContext)
+    const [savingRegexValue, setSavingRegexValue] = useState(false)
+    const selectedCIPipeline = props.filteredCIPipelines?.find((_ciPipeline) => _ciPipeline?.id === props.pipelineId)
+
+    const isValidRegex = (pattern: string, value: string): boolean => new RegExp(pattern).test(value)
+
+    const _regexValue: Record<number, RegexValueType> = props.material.reduce(
+        (acc, mat) => {
+            acc[mat.gitMaterialId] = {
+                value: mat.value,
+                isInvalid: mat.regex ? !isValidRegex(mat.regex, mat.value) : false,
+            }
+            return acc
+        },
+        {} as Record<number, RegexValueType>,
+    )
+
+    const [regexValue, setRegexValue] = useState<Record<number, RegexValueType>>(_regexValue)
```

Comment:
would the regexValue change on change of _regexValue

Reply:
Yes

Reply:
Then we would need useEffect, right?